### ISSUE TITLE
Feature: Add setting to always auto size details layout columns to fit

### ISSUE THREAD

**Issue by welles** [files-community/Files#9377]
### What's the Problem?

Like in the native file explorer, you have to manually resize columns or click 'Size all columns to fit' when working with very long / short file names

**Requirements**
- Add setting to always auto size columns to fit
- Entering a directory triggers auto size
- Adding / removing a file or folder triggers auto size
- Renaming a file or folder triggers auto size

### Alternatives

There could be a hotkey to 'resize to fit' all columns in the current view, which would already be an improvement over having to right-click the column header bar and select 'Size All Columns To Fit'

**Comment by yaichenbaum** [comment]
If there was an option to always auto size columns and you overrode a columns width, would you expect it to auto size the next time you navigated to the same directory?

**Comment by intergrav** [comment]
+1, I keep running into this issue where there is a big empty space in the file type area / name and it drives me crazy lol

![image](https://user-images.githubusercontent.com/42325132/196003309-d41ec4ef-672b-4bf7-87d9-9aac4ea5d894.png)

![image](https://user-images.githubusercontent.com/42325132/196003268-15f3120e-1e2b-4ef3-aba9-1e13a5798c8c.png)

You could fit a lot more with this

**Comment by yaichenbaum** [comment]
@intergrav we can solve that independently from this request by chaning the default width of the type column.

**Comment by yaira2** [comment]
> Adding / removing a file or folder triggers auto size
Renaming a file or folder triggers auto size

These can affect performance, I wonder if a hot key is the better way to do this.

**Comment by Draise14** [comment]
Yeah, this too would be very nice. I found the unresponsive column width loosing a lot of space specially on widescreens.

**Comment by yaira2** [comment]
> If there was an option to always auto size columns and you overrode a columns width, would you expect it to auto size the next time you navigated to the same directory?

I'd like to wrap up the planning on this issue, but before this can be done, we need to finalize the behavior for the above-mentioned scenario.

**Comment by rhiinschnookele** [comment]
I would expect it to auto size the next time I navigate to the same
directory.

Thank you!

Am 31.10.2024 16:53 schrieb Yair: 

>> If there was an option to always auto size columns and you overrode a columns width, would you expect it to auto size the next time you navigated to the same directory?
> 
> I'd like to finalize this feature, but before this can be done, we need to settle on the behavior for the above-mentioned scenario. 
> 
> --
> Reply to this email directly, view it on GitHub [1], or unsubscribe [2].
> You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>
 

Links:
------
[1]
https://github.com/files-community/Files/issues/9377#issuecomment-2450228648
[2]

**Comment by ELowry** [comment]
> > If there was an option to always auto size columns and you overrode a columns width, would you expect it to auto size the next time you navigated to the same directory?
> 
> I'd like to finalize this feature, but before this can be done, we need to settle on the behavior for the above-mentioned scenario.

I can see two valid approaches to this:
1. Changing a column width changes it across the whole app (probably as a percentage width).
2. Changing a column width only affects the current folder, and is saved locally.

The first option is great for consistency and muscle memory (most people will "set & forget"), but can be a pain for people who have things like GIT folders that will have more columns than usual, and may start acting differently.
The second one can be great for power-users, but would likely require a button or something to "save current column widths as default" so that the user can override the defaults.


Back to my earlier message, this could also all be superfluous if the "default" layout is smart enough; as I was describing with using "[minimum widths](https://github.com/files-community/Files/issues/9377#issuecomment-1175533319)" to ensure all columns can display what they need; and then anything left (file name, git commit messages, etc.) are given a default value that determines how much of the remaining space they try to take over (and their minimum width as well, leading to overflow, naturally).

Depending on how hard these things are to implement, this could be worth A/B testing with some temporary option in pre-release?

**Comment by rhiinschnookele** [comment]
I would agree to "Depending on how hard these things are to implement,
this could be worth A/B testing with some temporary option in
pre-release?"

Am 01.11.2024 12:48 schrieb Eric Lowry: 

> If there was an option to always auto size columns and you overrode a columns width, would you expect it to auto size the next time you navigated to the same directory? 
> 
> I'd like to finalize this feature, but before this can be done, we need to settle on the behavior for the above-mentioned scenario.

I can see two valid approaches to this: 

 	* Changing a column width changes it across the whole app (probably as
a percentage width).
 	* Changing a column width only affects the current folder, and is
saved locally.

The first option is great for consistency and muscle memory (most people
will "set & forget"), but can be a pain for people who have things like
GIT folders that will have more columns than usual, and may start acting
differently.
 The second one can be great for power-users, but would likely require a
button or something to "save current column widths as default" so that
the user can override the defaults. 

Back to my earlier message, this could also all be superfluous if the
"default" layout is smart enough; as I was describing with using
"minimum widths [1]" to ensure all columns can display what they need;
and then anything left (file name, git commit messages, etc.) are given
a default value that determines how much of the remaining space they try
to take over (and their minimum width as well, leading to overflow,
naturally). 

Depending on how hard these things are to implement, this could be worth
A/B testing with some temporary option in pre-release? 

--
Reply to this email directly, view it on GitHub [2], or unsubscribe [3].
You are receiving this because you are subscribed to this thread.Message
ID: ***@***.***> 

Links:
------
[1]
https://github.com/files-community/Files/issues/9377#issuecomment-1175533319
[2]
https://github.com/files-community/Files/issues/9377#issuecomment-2451747565
[3]

**Comment by yaira2** [comment]
What if a user overrode the width to be small, and then added a new file with a large name? Should the columns resize?

**Comment by ELowry** [comment]
> What if a user overrode the width to be small, and then added a new file with a large name? Should the columns resize?

That would be the point of "minimum widths" based on the type of column IMHO. Though there's an argument to be made for respecting user choices, even if they make no immediate sense to the designer.

**Comment by yaira2** [comment]
@ELowry, I might be missing the connection between minimum widths and this feature request. One of the requirements mentioned by the original author is that adding or renaming a file should automatically resize the columns. My question is whether this requirement should still apply when the user has manually set the column sizes for a specific directory. I hope this clarifies my question.

**Comment by ELowry** [comment]
> [@ELowry](https://github.com/ELowry), I might be missing the connection between minimum widths and this feature request. One of the requirements mentioned by the original author is that adding or renaming a file should automatically resize the columns. My question is whether this requirement should still apply when the user has manually set the column sizes for a specific directory. I hope this clarifies my question.

*Edit: sorry, this got rambley, but these things are hard to communicate in writing.*

Indeed; I was mostly focusing on the core idea of auto-resizing, and hadn't really re-read the initial requirements.


My personal stance on UX design is that it is not possible to ensure that everything is always displayed: file names can get massive, and having the name field auto-resize whenever a file name is edited is likely to lead to situations where all other columns get squashed.
I believe that the generally accepted UX design approach would be to focus on finding a good balance between the UI adapting to the width of the window, while ensuring maximum readability across a wide range of window widths and screen resolutions.

IMHO:

The file name tends to be the one that takes up the most space; so it would stand to reason to try and "optimize" all other columns to be as narrow as necessary, and give file names all available space.
This leaves 2 "problematic" scenarios:
* Below a certain width, it is likely to be impossible to properly fit all columns while still leaving space for the file name field. In this case, there are multiple possible approaches that can be taken, including "squashing" other columns some to try and keep the file name column readable; but also simply overflowing (as is currently the case). [this is the bit where a "minimum width" comes into play]
* For larger monitors, wider views could be hard to read if the file names are too far away from the other columns (if it's 60% of the whole width, and file names are short, it could be hard to associate the file names with the next column visually). So it could make sense to start scaling up the "secondary" columns' widths to "spread out" the information horizontally, which would keep things visible.

I think this second issue corresponds to when the initial request mentioned "short names".

**Comment by yaira2** [comment]
The "size to fit" feature isn't related to the window size, rather, it resizes the columns to fit the text.

**Comment by dmorris68** [comment]
It feels like this request was intended for Details layout, so not sure if my request fits here, but it is related to auto-sizing column widths. 

I'm more interested in List mode column widths. I pretty much live in List view mode, and much prefer Windows 10/11 default behavior of automatically sized List mode columns. Here's a Windows Explorer screenshot -- notice the long filenames sorted to the front make the first column very wide, but the subsequent columns of shorter folder names are quite narrow. This optimizes whitespace while still showing complete filenames.

![Image](https://github.com/user-attachments/assets/f3a4f475-c84a-4882-8da5-2f7d533bf56c)

Now compare the same folder shown in Files:

![Image](https://github.com/user-attachments/assets/47068671-82fe-41ee-bb55-c8173ec667af)

Here with the fixed column width you simultaneously have cut-off filenames in the left column, and excess whitespace in the other columns. Even more frustrating, unless I'm missing something, there's no way to manually resize the columns in List mode, so I'm left to hovering over the cut-off filenames and waiting for a popup to see the complete filename (in tiny hover print). Just not a good experience for me.

**Comment by yaira2** [comment]
> It feels like this request was intended for Details layout, so not sure if my request fits here, but it is related to auto-sizing column widths.

@dmorris68 these are different requests, however you can refer to #1033 for more details.

**Comment by yair100** [comment]
@welles this feature will be included in the next release.
